### PR TITLE
Feature/backend/swap courses

### PR DESCRIPTION
**Pull request by SamNormcoreWayne**
Implemented backend for SwapCourse.
~TODO: Testing for each route.~
Testing for GET is done

### CODE REVIEW

File: backend/routes/courses/swap/accecptSwapCourses.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+const DB_BASE_URL = env('DB_BASE_URL')
+
+/**
+ * @typedef ErrorResponse
+ * @property {[integer]} statusCode
+ * @property {[string]} message
+ */
+
+/**
+ * @typedef SwapeeRequest
```
Comment (SamNormcoreWayne) [inline backend/routes/courses/swap/accecptSwapCourses.js]: review required

File: backend/routes/courses/swap/accecptSwapCourses.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * @security JWT
+ */
+
+module.exports = function (req, res, next) {
+  const swapReq = {
+    'userId': req.body.userId,
+    'courseToGiveId': req.body.courseToGiveId,
+    'courseToGetId': req.body.courseToGetId
+  }
+  return axios.post(urlJoin(DB_BASE_URL, 'courses', 'swap', 'accept'), swapReq)
```
Comment (CCorrado) [inline backend/routes/courses/swap/accecptSwapCourses.js]: `swap` should be `swaps`
image

File: backend/routes/courses/swap/accecptSwapCourses.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * @route POST /courses/swaps/accept
+ * @group Courses
+ * @param {SwapCourses.model} SwapCourses.body.required - A swap req contains userId and courseToGiveId, courseGetId, swapeeAccept
+ * @return {CourseSwapResponse.model} 201 - Accept successfully
+ * @return {ErrorResponse.model}  default - HttpError - Swap cannot created
+ * @security JWT
+ */
+
+module.exports = function (req, res, next) {
+  const swapReq = {
```
Comment (CCorrado) [inline backend/routes/courses/swap/accecptSwapCourses.js]: this is the `acceptSwapRequest` which requires the following:
```
courseSwapId -- ID of the swap (you would have gotten this from the GET call to /swaps)
swapeeUserId -- the ID of the user accepting the swap (check to see if it is different from the swaperUserId, and also check to see if it is the same as the user making the request)
swapeeAccept -- true or false if this swap has been accepted or declined
```

File: backend/routes/courses/swap/createSwapCourses.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ * @security JWT
+ */
+
+module.exports = function (req, res, next) {
+  const swapReq = {
+    'userId': req.body.userId,
+    'courseToGiveId': req.body.courseToGiveId,
+    'courseToGetId': req.body.courseToGetId
+  }
+  return axios.post(urlJoin(DB_BASE_URL, 'courses', 'swap', 'create'), swapReq)
```
Comment (CCorrado) [inline backend/routes/courses/swap/createSwapCourses.js]: `swap` should be `swaps`

File: backend/routes/courses/swap/createSwapCourses.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+ * @group Courses
+ * @param {SwapCourses.model} SwapCourses.body.required - A swap req contains userId and courseToGiveId, courseGetId
+ * @return {CourseSwapResponse.model} 201 - Create successfully
+ * @return {ErrorResponse.model}  default - HttpError - Swap cannot created
+ * @security JWT
+ */
+
+module.exports = function (req, res, next) {
+  const swapReq = {
+    'userId': req.body.userId,
```
Comment (CCorrado) [inline backend/routes/courses/swap/createSwapCourses.js]: `userId` should be `swaperUserId`

File: backend/tests/unit/controllers/courses/courses.swap.test.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    try {
+      return request(app).get('/api/courses/swaps').set('Authorization', 'Bearer someUniqueToken').expect(400)
+    } catch (e) {
+      assert(false, e.message)
+    }
+  })
+
+  it('Should successfully return swap course list to the user', () => {
+    nock('http://cooper-database-api:8080/').get('/courses/swaps').delay(100).reply(200, {
+      'courseSwapId': '0',
```
Comment (CCorrado) [inline backend/tests/unit/controllers/courses/courses.swap.test.js]: this reply should be an array of objects as opposed to a single object

Comment (SamNormcoreWayne) [reply backend/tests/unit/controllers/courses/courses.swap.test.js]: changed `{...}` into `[{...}]`

**Pull request merged**